internal/policy/kubernetes_client.go

```diff
@@ -64,10 +66,47 @@ func createControllerRuntimeClient() (client.Client, error) {
 
 func (k *kubernetes) fetchEnterpriseContractPolicy(ctx context.Context, name types.NamespacedName) (*ecp.EnterpriseContractPolicy, error) {
 	policy := &ecp.EnterpriseContractPolicy{}
+	if name.Namespace == "" {
+		namespace, err := getCurrentNamespace()
+		if err != nil {
+			return nil, err
+		}
+		name.Namespace = namespace
+	}
 	err := k.client.Get(ctx, name, policy)
 	if err != nil {
 		return nil, err
 	}
 
 	return policy, nil
 }
+
+// getCurrentNamespace returns the namespace of the current context if one is set.
+func getCurrentNamespace() (namespace string, err error) {
+	baseErr := "Unable to determine current namespace"
```

Comment:
Nit: go errors start with [lowercase letters](https://github.com/golang/go/wiki/CodeReviewComments#error-strings), it is weird but a go thing

```suggestion
	baseErr := "unable to determine current namespace"
```

Reply:
Thanks for the tip!